cmake/toolchains/clang.cmake

```diff
@@ -1,14 +1,4 @@
 
 set(CMAKE_C_COMPILER clang)
 set(CMAKE_CXX_COMPILER clang++)
-
-option(CLANG_TIDY "Enable clang-tidy static analysis" OFF)
-if(CLANG_TIDY)
-    find_program(CLANG_TIDY_EXE NAMES "clang-tidy")
-    if(CLANG_TIDY_EXE)
-        message(STATUS "Found clang-tidy: ${CLANG_TIDY_EXE}")
-        set(CMAKE_CXX_CLANG_TIDY "${CLANG_TIDY_EXE}")
-    else()
-        message(WARNING "clang-tidy not found!")
-    endif()
-endif()
\ No newline at end of file
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
```

Comment:
From [the docs](https://cmake.org/cmake/help/latest/variable/CMAKE_EXPORT_COMPILE_COMMANDS.html):
> Note This option is implemented only by [Makefile Generators](https://cmake.org/cmake/help/latest/manual/cmake-generators.7.html#makefile-generators) and [Ninja Generators](https://cmake.org/cmake/help/latest/manual/cmake-generators.7.html#ninja-generators). It is ignored on other generators.

I guess this should either be in:
1. gen:ninja target, or
2. Directly inline in CMakeLists.txt to force it on even preset-less builds